feature/signup

nahokyun opened the pull request:
회원가입 기능을 구현하였습니다
immutable object의 사용이유를 지키기 위해 @Builder 사용으로 set함수를 제거해주었습니다.

nahokyun committed:
feature/signup
회원가입 기능을 구현하였습니다
immutable object를 지키기 위해 @Builder 사용으로 set함수를 제거해주었습니다.

sunwootest commented on src/main/java/study/movieservice/controller/MemberController.java:
```diff
@@ -0,0 +1,19 @@
+public class MemberController {
+    private final MemberService memberService;
+
+    @PostMapping("/join")
+    public String join(@RequestBody Member member){
+        memberService.save(member);
+        return "ok";
```
회원가입을 성공했다는 것으로 ok를 return할 필요가 있을까요? 200 http 상태코드면 충분해 보입니다.

nahokyun replied:
ResponseEntity를 통해 HttpStatus.OK를 리턴하는 방법으로 대체하였습니다.

sunwootest replied:
아무것도 return 하지 않으면 어차피 200 http 상태코드를 보내줄텐데 다른 방법도 있으니 한 번 알아보세요 :)

sunwootest commented on src/main/java/study/movieservice/domain/Member.java:
```diff
@@ -0,0 +1,29 @@
+import static study.movieservice.domain.Grade.BASIC;
+
+
+@Getter
+
```
사소한 공백도 신경써주세요! 팀원과 함께 코드 컨벤션에 대해 깊게 고민하고 어떻게 자동화 할지 고민하면 좋겠습니다. 이 부분은 나중에 처리하려면 일이 커지니 프로젝트 초반부에 정하는 것이 좋습니다.

nahokyun replied:
팀원과 상의후 통상적으로 사용되는 방향으로 컨벤션을 정해 진행하도록 하겠습니다.

sunwootest commented on src/main/resources/application.properties:
```diff
@@ -1,7 +1,6 @@
-#spring.datasource.url =
-#spring.datasource.username=
-#spring.datasource.password=
+spring.datasource.url=
+spring.datasource.username=
+spring.datasource.password=
 
 #MyBatis xml ???? resultType? ???? ???? ?? ??. ( domian ??? ?? ?? )
 mybatis.type-aliases-package=study.movieservice.domain
```
git ignore 작업해주세요

nahokyun replied:
application.properties를 gitignore에 추가하였습니다.

sunwootest commented on src/main/java/study/movieservice/repository/mybatis/MybatisMemberRepository.java:
```diff
@@ -0,0 +1,23 @@
+    }
+
+    @Override
+    public Optional<Member> findByloginId(String loginId) {
+        return memberMapper.findByloginId(loginId);
```
카멜 케이스가 안지켜진 것 같네요

nahokyun replied:
findByLoginId로 바꾸어 카멜케이스를 지키도록 수정하였습니다.

sunwootest commented on src/main/java/study/movieservice/repository/mybatis/MybatisMemberRepository.java:
```diff
@@ -0,0 +1,23 @@
+import study.movieservice.domain.Member;
+
+import java.util.Optional;
+
+@Repository
+@RequiredArgsConstructor
+public class MybatisMemberRepository implements MemberMapper {
+    private final MemberMapper memberMapper;
```
Mapper와 기능이 완전히 동일한데 Repository를 굳이 다시 만들어줄 필요가 없어보입니다.

nahokyun replied:
Mapper에 대한 이해가 부족했던것같습니다. 불필요했던 MybatisMemberRepository 파일을 제거하였습니다.

sunwootest reviewed (commented):
공백 같은 코딩 컨벤션에 대한 지적할 것들이 좀 있어 보입니다. 이 부분을 개선해주세요. 지금 당장 개선하기 어려우면 인텔리제이 자동정렬 기능도 활용해보시고 그 이후에 팀원과 꼭 협의해서 코딩컨벤션을 지켜주세요.

nahokyun committed:
feature/signup
불필요한 repository파일 삭제,
camel case를 반영한 메소드 이름조정
ok문자열 리턴에서 상태코드 출력으로 변경 진행하였습니다.

nahokyun committed:
Update application.properties

sunwootest commented on src/main/java/study/movieservice/controller/MemberController.java:
```diff
@@ -0,0 +1,21 @@
+    @PostMapping("/join")
+    public ResponseEntity<Void> join(@RequestBody Member member){
+        memberService.save(member);
+        return new ResponseEntity<>(HttpStatus.OK);
```
200 코드만 보내줄거면 return 값을 void로 처리하는 것이 더 간단해 보입니다 :)

nahokyun replied:
저장시 200 코드만을 응답하므로 ResponseEntity대신 void로 처리하여 좀더 간결하게 변경하였습니다.

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -0,0 +1,23 @@
+    private final MemberMapper memberMapper;
+
+    public void save(Member member){
+        Member member1= Member.builder()
```
member1 member2와 같은 네이밍은 지양해야합니다. 좋은 변수명은 어떻게 지어야하는지 고민하면 좋겠습니다 :)

nahokyun replied:
변수의 목적을 확실하게 나타내기위해 encryptMember로 변경하였습니다.

sunwootest reviewed (commented):
리뷰 반영했으면 코멘트 남겨주세요.

nahokyun committed:
feature/signup_na
member1로 선언한 변수명을 좀더 의도를 확실하게 하기위해 encryptMember로 변경하였고 ResponseEntity를 void로 변경하여 코드의 가독성을 높히도록 변경하였습니다.

nahokyun commented:
코드의 가독성을 위한 변수명 변경과 마찬가지로 가독성을 위해 메소드의 자료형을 변경하였습니다.

sunwootest commented:
임시 approve 드립니다 merge는 하지 말아주세요 :)
